docs/spec.md

```diff
@@ -5092,32 +5433,45 @@ def sign(x):
     if compare(x, 0, EQ, SIGNED): return 0
     return 1
   elif is_float(x):
-    if x is NaN: return NaN
-    if compare(x, -0.0, EQ, FLOAT): return -0.0
-    if compare(x, +0.0, EQ, FLOAT): return +0.0
-    if compare(x, 0.0, LT, FLOAT): return -1.0
-    return 1.0
+    def sign_computaion_float_type(x):
+        if x is NaN: return NaN
+        if compare(x, -0.0, EQ, FLOAT): return -0.0
+        if compare(x, +0.0, EQ, FLOAT): return +0.0
+        if compare(x, 0.0, LT, FLOAT): return -1.0
+        return 1.0
+    sign_computaion_float_type(x)
```

Comment:
I don't think we need this change. Below, we can just say `dequantize_op_quantize(sign, x, type(result))`.